### DISCUSSION TITLE
Importing a Tailwind + Vue component library in another project

### DISCUSSION THREAD

**Discussion by scmmishra** [tailwindlabs/tailwindcss#3742] [Help] [answered]
My team has a component system called **Zeal** we've built on Vue and Tailwind, that is **published as a package** and used across our products and websites. 

The **projects we use Zeal in also use tailwind**, so we assume that the tailwind classes in the components will be applied directly in all projects we use Zeal. And this works well in the development environment where PurgeCSS is disabled. 

But, when we actually build the project for production, PurgeCSS keeps all the styles that is present in the project files and purges styles used Zeal. An example would be

```vue
// Project using ZButton Component from Zeal
<div class="container mx-auto">
  <z-button>Hello, World</z-button>
</div>
```
```vue
// ZButton defined in Zeal
<button class="px-4 py-2 bg-blue-200 hover:bg-blue-300 rounded-md">
  <slot></slot>
</div>
```

When building the page, only `container` and `mx-auto` would remain, the rest would be purged.

Is there a way for me to whitelist an external library that uses tailwind as well? Or am I doing something wrong here? I understand this may not be a question directly related to tailwind, however I'd appreciate the help. 

P.S. When building the Zeal we generate a few css files (`tailwind.css`,  `typography.css` and `layout.css`). However we do not import them in our projects directly.

**Comment by rtkaaho** [comment, marked as answer]
Ahhh, we might be having similar issues. I just made this post aswell. https://github.com/tailwindlabs/tailwindcss/discussions/3747

**Reply by scmmishra** [reply]
Yeah, seems like it's the same problem, I'll try the solution suggested.
